src/Windows/Markdown/Editor/Impl/Commands/InsertRCodeBlock.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.Services;
+using Microsoft.Common.Core.UI.Commands;
+using Microsoft.Languages.Editor.Controllers.Commands;
+using Microsoft.Markdown.Editor.ContentTypes;
+using Microsoft.R.Editor;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.Markdown.Editor.Commands {
+    internal sealed class InsertRCodeBlock : ViewCommand {
+        private readonly IREditorSettings _settings;
+        public InsertRCodeBlock(ITextView textView, IServiceContainer services)
+            : base(textView, new CommandId(MdPackageCommandId.MdCmdSetGuid, MdPackageCommandId.icmdInsertRCodeBlock), false) {
+            _settings = services.GetService<IREditorSettings>();
+        }
+
+        public override CommandStatus Status(Guid group, int id) {
+            if (!TextView.TextBuffer.ContentType.TypeName.EqualsOrdinal(MdProjectionContentTypeDefinition.ContentType)) {
+                return CommandStatus.Invisible;
+            }
+            return TextView.IsCaretInRCode() ? CommandStatus.Supported : CommandStatus.SupportedAndEnabled;
```

Comment:
Do we have any tests for `IsCaretInRCode`?

Reply:
Good question. Let me check markdown tests

Reply:
Trying to update xunit to 2.3.1

Reply:
Does not help. https://developercommunity.visualstudio.com/content/problem/165706/unit-tests-not-found.html